internal/dataplane/dataplane_config.go

```diff
@@ -41,14 +44,18 @@ func (i *GetDataplaneConfigJSONInput) GetDataplaneConfigJSON() ([]byte, error) {
 		},
 		XDSServer: XDSServerConfig{
 			Address: localhostAddr,
-			Port:    20000,
```

Comment:
Envoy complains that 20000 is already in use whenever it starts in the current state. It looks like dataplane picks a free port if we don't set it from the control plane.

Reply:
🤔 I'm wondering if this is going to work in that case:

* Consul Dataplane [starts an xDS server/proxy](https://github.com/hashicorp/consul-dataplane/blob/9d470c739e3eeb0a4555c452482b50acb376227c/pkg/consuldp/xds.go#L56)

Then it configures an Envoy cluster to point at that xDS server:

* https://github.com/hashicorp/consul-dataplane/blob/9d470c739e3eeb0a4555c452482b50acb376227c/pkg/consuldp/bootstrap.go#L59
* https://github.com/hashicorp/consul-dataplane/blob/9d470c739e3eeb0a4555c452482b50acb376227c/internal/bootstrap/bootstrap_tpl.go#L240

I think if the Envoy `socket_address.port_value = 0` then it wouldn't point at the xDS server started by Consul Dataplane that is running on a randomly selected port.

Do you know if it will work? I might be missing something.

Reply:
It does work because `socker_address.port_value` gets the value from the XDS listener address and not the `xdsServer.BindPort` present in dataplane config.

https://github.com/hashicorp/consul-dataplane/blob/9d470c739e3eeb0a4555c452482b50acb376227c/pkg/consuldp/bootstrap.go#L84